solana/program_v2/programs/gateway_v2/src/instructions/network/gatekeeper_withdraw.rs

```diff
@@ -47,21 +48,17 @@ pub fn gatekeeper_withdraw(ctx: Context<GatekeeperWithdrawAccount>, mut amount:
 pub struct GatekeeperWithdrawAccount<'info> {
     #[account(
         seeds = [GATEKEEPER_SEED, authority.key().as_ref(), gatekeeper.gatekeeper_network.key().as_ref()],
-        bump = gatekeeper.gatekeeper_bump
+        bump = gatekeeper.gatekeeper_bump,
+        constraint = gatekeeper.can_access(&authority, GatekeeperKeyFlags::WITHDRAW) @ GatekeeperErrors::InsufficientAccessAuthKeys,
     )]
     pub gatekeeper: Account<'info, Gatekeeper>,
-    #[account()]
-    pub authority: Signer<'info>,
-    // TODO: I think we don't need this
-    pub system_program: Program<'info, System>,
     pub spl_token_program: Program<'info, Token>,
-    // TODO: THIS COMMENT IS NOT RIGHT
-    // Verification is done in code
-    // TODO: DO Verification here. That requires to pass and deserialize the network.
-    pub mint_account: Account<'info, Mint>,
+    pub authority: Signer<'info>,
     #[account(mut)]
     pub receiver_token_account: Account<'info, TokenAccount>,
-    // TODO: CHECK: the owner of the gatekeeper_token_account should be the gatekeeper
-    #[account(mut)]
+    #[account(
+    mut,
```

Comment:
Here gatekeeper_token_account is labeled "mut" as mentioned above

Reply:
See comment above

Reply:
👍